cli/pkg/config/happy_version_lock.go

```diff
@@ -0,0 +1,92 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+type HappyVersionLockFile struct {
+	HappyVersion string
+	Path         string `json:"-"`
+}
+
+func NewHappyVersionLockFile(projectRoot string, requiredVersion string) (*HappyVersionLockFile, error) {
+
+	if projectRoot == "" {
+		return nil, errors.New("No projectRoot specified")
+	}
+
+	if requiredVersion == "" {
+		return nil, errors.New("No requiredVersion specified")
+	}
+
+	path := calcHappyVersionPath(projectRoot)
+
+	return &HappyVersionLockFile{
+		HappyVersion: requiredVersion,
+		Path:         path,
+	}, nil
+}
+
+func DoesHappyVersionLockFileExist(projectRoot string) bool {
+	filePath := calcHappyVersionPath(projectRoot)
+	_, err := os.Stat(filePath)
+	return err == nil
+}
+
+func LoadHappyVersionLockFile(projectRoot string) (*HappyVersionLockFile, error) {
+
+	filePath := calcHappyVersionPath(projectRoot)
+
+	versionFile, err := os.Open(filePath)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to open happy version lock file")
+	}
+	defer versionFile.Close()
+
+	contents, err := ioutil.ReadAll(versionFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to read happy version lock file")
+	}
+
+	hvlf := HappyVersionLockFile{}
+
+	err = json.Unmarshal(contents, &hvlf)
```

Comment:
```suggestion
	hvlf := HappyVersionLockFile{}
	err = json.NewDecoder(versionFile).Decode(&hvlf)
```

when I already have a `reader` interface like a file, I opt for this method of decoding JSON because its all done in one step and I don't have to handle the the io.ReadAll errors. Its also a nice trick when you want to decode a stream of bytes like on a network request.